flytekit-java/src/main/java/org/flyte/flytekit/SdkBindingData.java

```diff
@@ -88,63 +88,134 @@ private static <T> SdkBindingData<T> ofPrimitive(Function<T, Primitive> toPrimit
   }
 
   public static <T> SdkBindingData<List<T>> ofCollection(
-      List<T> collection, Function<T, SdkBindingData<T>> function) {
+      List<T> collection, LiteralType literalType, Function<T, SdkBindingData<T>> function) {
     return SdkBindingData.ofBindingCollection(
-        collection.stream().map(function).collect(Collectors.toList()));
+        literalType, collection.stream().map(function).collect(Collectors.toList()));
+  }
+
+  private static <T> SdkBindingData<List<T>> ofTCollection(
+      List<T> collection, LiteralType literalType, Function<T, BindingData> bindingDataFn) {
```

Comment:
```suggestion
  interface ToBindingDataFunction<T> {
    BindingData apply(T value);
  }
      List<T> collection, LiteralType literalType, ToBindingDataFunction bindingDataFn) {
```
```suggestion
  private static <T> SdkBindingData<List<T>> ofTCollection(
```
You had to change the name of the method because its eraused firm is the same as the other. But you gave it a name that it is applicable to both methods, both receive a `List<T>`

I think that if we introduce our own interface we can break the erauser problem and keep using the `ofCollection` name